fMRIVolume/scripts/MotionCorrection.sh

```diff
@@ -44,6 +59,25 @@ case "$MotionCorrectionType" in
     ;;
 esac
 
+
+# Run nonlinear registration if needed
+
+if [ "${ReferenceReg}" == "nonlinear" ] ; then
+  verbose_echo " ... computing nonlinear transform to reference"
+  verbose_echo "     ... generating bold average"
+  ${FSLDIR}/bin/fslmaths ${WorkingDirectory}/${OutputfMRIBasename} -Tmean ${WorkingDirectory}/${OutputfMRIBasename}_avg
+
+  verbose_echo "     ... running fnirt: fnirt --in=${WorkingDirectory}/${OutputfMRIBasename}_avg --ref=${Scout} --iout=${WorkingDirectory}/${OutputfMRIBasename}_avg_nonlin --cout=${WorkingDirectory}/mc2ref_warp"
+  ${FSLDIR}/bin/fnirt --in=${WorkingDirectory}/${OutputfMRIBasename}_avg --ref=${Scout} --iout=${WorkingDirectory}/${OutputfMRIBasename}_avg_nonlin --cout=${WorkingDirectory}/mc2ref_warp
```

Comment:
I noticed that this `fnirt` call isn't using either `--aff` or `--regmask`...

Reply:
There is no affine transform needed, as the timeseries has already been registered linearly to the external scout, so this is an additional non-linear registration. But I have added the reference scout mask.

Reply:
I guess the preferred FSL way of using FNIRT is to specify the original image and an affine rather than an affinely registered image.  I don't know how much this matters though.

Reply:
@glasserm To use the `--aff` we would need to then first compute the motion corrected version of bold to its first image or scout and then compute an affine transform to reference scout and then (optionally) also add the nonlinear transform. Would this additional two step path improve FNIRT solution compared to working with BOLD that has been directly motion registered to reference Scout?

Reply:
--aff would be whatever linear transform is computed before the nonlinear transform.  It would be atypical to start with a nonlinear transform first.

Reply:
But, per your earlier comment, acceptable not to use `--aff` here, given that the data has already been linearly registered through the motion correction, right?

BTW: It has never been clear to me from the FNIRT documentation whether the `--aff` input is the _only_ affine that FNIRT ever uses, or if it is just the "starting guess" (and then iterates to update the affine component as part of iterating over the nonlinear warps).  Several parts of the documentation explicitly refer to it as the "starting guess", but other parts lead me to wonder if the affine component is ever indeed updated.

Reply:
Yes, though see my proposal for a change in how this is done.

Reply:
That seems an empirical question, and unless we have a good theoretical reason to favor that approach, introducing a "scout" for the run that is separate from the "reference scout" seems unnecessarily complicated, and would prevent the simplification that I've proposed that we simply treat the `--fmriref` as the only "scout".

Reply:
@mharms a warpfield can encode an affine, and I doubt fnirt does anything to prevent the warpfield from developing some amount of global affine component, and I really doubt that fnirt makes any modification to any actual set of affine parameters while registering.  Some registration tools basically pretend the data has already been resampled according to the initial input guess (in reality, using the initial affine as a premat when sampling through the current warpfield guess), which means you would need to concatenate the warpfield onto the initial affine to get the correct warp, and I think fnirt might be among them.  Do any of these get at your question?

Reply:
I think there is a preference in FSL to feed FNIRT the original image and the affine matrix rather than the resampled image and no affine.